File: BMIApp/app/src/main/java/pl/pjwstk/bmiapp/RecipesFragment.java
package pl.pjwstk.bmiapp;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

public class RecipesFragment extends Fragment {

    private LinearLayout recipesContainer;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_recipes, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        recipesContainer = view.findViewById(R.id.recipesContainer);

        // Załadowanie przykładowych przepisów
        loadRecipes();
    }

    /**
     * Załadowanie przepisów do UI
     */
    private void loadRecipes() {
        Recipe[] recipes = Recipe.getSampleRecipes();

        for (Recipe recipe : recipes) {
            View recipeView = createRecipeView(recipe);
            recipesContainer.addView(recipeView);
        }
    }

    /**
     * Tworzenie widoku dla pojedynczego przepisu
     */
    private View createRecipeView(Recipe recipe) {
        View recipeView = getLayoutInflater().inflate(R.layout.item_recipe, recipesContainer, false);

        // Ustawienie tytułu przepisu
        TextView titleTextView = recipeView.findViewById(R.id.recipeTitleTextView);
        titleTextView.setText(recipe.getTitle());

        // Ustawienie kalorii
        TextView caloriesTextView = recipeView.findViewById(R.id.recipeCaloriesTextView);
        caloriesTextView.setText(String.format(getString(R.string.recipe_calories), recipe.getCalories()));

        // Ustawienie składników
        TextView ingredientsTextView = recipeView.findViewById(R.id.recipeIngredientsTextView);
        StringBuilder ingredientsBuilder = new StringBuilder();
        for (String ingredient : recipe.getIngredients()) {
            ingredientsBuilder.append("• ").append(ingredient).append("\n");
        }
        ingredientsTextView.setText(ingredientsBuilder.toString());

        // Ustawienie instrukcji
        TextView instructionsTextView = recipeView.findViewById(R.id.recipeInstructionsTextView);
        instructionsTextView.setText(recipe.getInstructions());

        return recipeView;
    }
}

File: BMIApp/app/src/main/java/pl/pjwstk/bmiapp/ui/fragments/recipes/RecipesFragment.kt
package pl.pjwstk.bmiapp.ui.fragments.recipes

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.LinearLayout
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.navigation.Navigation
import pl.pjwstk.bmiapp.R
import pl.pjwstk.bmiapp.data.models.Recipe
import pl.pjwstk.bmiapp.utils.NavigationHelper

class RecipesFragment : Fragment() {
    private val TAG = "RecipesFragment"
    private lateinit var recipesContainer: LinearLayout
    private lateinit var recipesList: Array<Recipe>

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_recipes, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        recipesContainer = view.findViewById(R.id.recipesContainer)

        recipesList = Recipe.getSampleRecipes()

        loadRecipes()
    }

    private fun loadRecipes() {
        for (recipe in recipesList) {
            val recipeView = createRecipeView(recipe)
            recipesContainer.addView(recipeView)
        }
    }

    private fun createRecipeView(recipe: Recipe): View {
        val recipeView = layoutInflater.inflate(R.layout.item_recipe, recipesContainer, false)

        val titleTextView: TextView = recipeView.findViewById(R.id.recipeTitleTextView)
        titleTextView.text = recipe.title

        val caloriesTextView: TextView = recipeView.findViewById(R.id.recipeCaloriesTextView)
        caloriesTextView.text = getString(R.string.recipe_calories, recipe.calories)

        val ingredientsTextView: TextView = recipeView.findViewById(R.id.recipeIngredientsTextView)
        val ingredientsBuilder = StringBuilder()
        for (ingredient in recipe.ingredients) {
            ingredientsBuilder.append("• ").append(ingredient).append("\n")
        }
        ingredientsTextView.text = ingredientsBuilder.toString()

        val instructionsTextView: TextView =
            recipeView.findViewById(R.id.recipeInstructionsTextView)
        instructionsTextView.text = recipe.instructions

        val generateShoppingListButton: Button =
            recipeView.findViewById(R.id.generateShoppingListButton)
        generateShoppingListButton.setOnClickListener {
            val navController =
                Navigation.findNavController(requireActivity(), R.id.nav_host_fragment)

            val recipeIndex = recipesList.indexOf(recipe)
            Log.d(TAG, "Generuję listę zakupów dla przepisu: ${recipe.title}, indeks: $recipeIndex")

            NavigationHelper.navigateToShoppingList(navController, recipeIndex)
        }

        return recipeView
    }
}